numpy/core/src/multiarray/ctors.c

```diff
@@ -1025,7 +1027,7 @@ PyArray_NewFromDescr_int(PyTypeObject *subtype, PyArray_Descr *descr, int nd,
              * the memory, but be careful with this...
              */
             memcpy(fa->strides, strides, sizeof(npy_intp)*nd);
-            sd *= size;
+            sd = size;
```

Comment:
Always good to be paranoid, just in case they are really after you :). I gave it a shot, though not quite sure about it.